src/app/mail/mail-list/mail-list.component.ts

```diff
@@ -44,32 +44,78 @@ export class MailListComponent implements OnInit, OnDestroy {
   }
 
   ngOnInit() {
+    this.route.params.subscribe(params => {
+      const mailFolderType: MailFolderType = params['folder'] as MailFolderType;
+      this.getMails( mailFolderType);
+    });
     this.store.select(state => state.mail).takeUntil(this.destroyed$)
       .subscribe((mailState: MailState) => {
         this.mails = mailState.mails;
       });
-    this.getMails();
 
   }
 
-  getMails() {
-    this.store.dispatch(new GetMails({ limit: 1000, offset: 0 }));
+  getMails(mailFolderType: MailFolderType) {
+    this.store.dispatch(new GetMails({ limit: 1000, offset: 0 , folder: mailFolderType }));
   }
 
-  rowSelectionChanged(event: any, mail: Mail) {
-    if (this.checkedMailIds.indexOf(mail.id) < 0) {
-      this.checkedMailIds = [...this.checkedMailIds, mail.id];
+  markSelectedMail(mail){
+    if (this.markedMailsMap.has(mail.id)) {
+      this.markedMailsMap.delete(mail.id);
     } else {
-      this.checkedMailIds = this.checkedMailIds.filter(checkedMailId => checkedMailId !== mail.id);
+      this.markedMailsMap.set(mail.id, mail);
     }
   }
 
-  moveToTrash() {
+  markAllMails(checkAll){
+    if (checkAll){
+      this.mails.forEach(mail => {
+        this.markedMailsMap.set(mail.id, mail);
+      })
+    } else {
+      this.markedMailsMap = new Map();
```

Comment:
Why not use the `clear` method of `Map` instead of creating new `Map`?

Reply:
fixed.